### ISSUE TITLE
Allow upload of editable vector topos illustrator photoshop powerpoint

### ISSUE THREAD

**Issue by brendanheywood** [theCrag/website#58]
Currently topos are drawn over the top of photos. We could also support the drawing of topos on a blank canvas, either inside thecrag by writing our own drawing tools, or outside the crag by leveraging formats like illustrator. This is most often used for overhead topo maps, or cliffs which are hard to photograph. The key thing we want to retain is the ability to collaborate on topos so no one person is a gatekeeper and there is no chance of an original file being lost. Enhancing the topo tool to become a general purpose drawing tool is potentially a lot of work, but could create the best outcome as all the topos would have a similar style (eg we could have predefined 'boulder' and 'cliff' styles, and we could update these styles for every topo in one go without going back and touching every one of them).

On the flip side allowing uploads of vector artwork is relatively easy and would give a lot of control to our power users, but essentially block less skilled users from contributing.

Tasks:
- [ ] Allow people to upload a set of raw vector files svg, illustrator, ppt
- [ ] Convert them to 'flat' simple vector drawing for serving
- [ ] Modify the topo tool to work with these files
- [ ] Importantly allow anyone to download the original file, modify it, and then
- [ ] re-upload it into the existing topo (this one has value even for raw photos)

**Comment by Drazhar** [comment]
I wouldn't take the effort of creating a vector drawing tool inside theCrag. You could never match the existing tools like Illustrator or even Inkscape. 
The idea to allow anyone to download the original file to modify on his own system and then re-upload it sounds great. 

Could be that some people don't want their vector drawings available to everyone to download as original. But in this case they can always upload it as a jpg as in the current state.

**Comment by brendanheywood** [comment]
Yeah I would never attempt to try and reproduce a generic drawing tool like illustrator, but I would happily invest the time to allow people to draw shapes which have meaning, eg 'this is a boulder outline', 'this is a track' and having the style determined by the system and not by the author. This fits much better with my long term vision of making consistent guides regardless of who made them, or how many people have contributed and regardless of skill level. Bringing the topo tool up to this level is definitely work but not actually that much work.

**Comment by brendanheywood** [comment]
> Could be that some people don't want their vector drawings available to everyone to download as original. But in this case they can always upload it as a jpg as in the current state.

This goes against the grain of our open licensing, we want the guide to be community data. If a topo isn't able to be license to us for ongoing modification then it's no good to us. It needs to be a living thing and kept up to date.

> edit: Because this would be an additional feature, it wouldn't block "less skilled users".

Yes it would, because if someone doesn't have illustrator or can't figure out how to use it then they can't fix a bug in an existing topo. It absolutely is a barrier to easy collaborative editing.

**Comment by brendanheywood** [comment]
Just documenting some more thoughts on this:

* [ ] to side step the issues with server side conversion to svg and possible a non optimal web version, we could simply allow a parallel upload of raw original and the final web ready version.
* [ ] long term how do we allow topos to be translated? Topos often contain highly curated typography and in one language. How can we support good typography and multi lang? Can we?
 * [ ] One solution is that all text is a topo is drawn on over the top using our tool and the topo contains no text at all. This is not great solution until #1614 is implemented. Best solution is we handle this same as elsewhere and inside the topo tool allow a language selector and can add the text and preview it in each lang.
 * [ ] another solution is we allow the raw vector topo to contain placeholder text which we can swap out server side. eg {{123456}} with a node id, and the server can replace this with the node name. The upside is the text style and thing like text along a path will be supported. The down side of this is we are restricted to node names, it's a bit harder for authors to create. I don't like this much at all.
